gcp/index.ts

```diff
@@ -0,0 +1,92 @@
+import * as k8s from "@pulumi/kubernetes";
+import * as gcp from "@pulumi/gcp";
+
+import * as config from "./config";
+import * as model from "./model";
+import * as gke from "./gke";
+import * as utils from "./utils";
+
+let cloud: model.Cloud;
+
+if (config.cloud == config.GcpCloud)
+  cloud = new gke.GcpCloud();
+else
+  throw new Error(`invalid cloud configuration: ${config.cloud}`);
+
+let cluster: model.KubernetesCluster = cloud.createKubernetesCluster();
+
+// K8s namespace for operator
+let namespaceName = config.operatorNamespace;
+
+// Output the cluster's kubeconfig and name
+export const kubeconfig = cluster.kubeconfig;
+export const clusterName = cluster.name;
+
+// Create a k8s namespace for operator
+let namespace = new k8s.core.v1.Namespace(namespaceName, {
+  metadata: {
+    // fixme: add to configuration, if DNE let pulumi generate random suffix?
+    // otherwise pulumi will append a random suffix to the namespace.. might be useful for integration testing to do that
+    name: namespaceName 
+  }
+}, {provider: cluster.k8sProvider});
+
+// Operator namespace name
+export const operatorNamespace = namespace.metadata.name;
+
+let serviceAccountName = utils.name("sa");
+let serviceAccount = cloud.operatorServiceAccount(
+  cluster, 
+  serviceAccountName, 
+  namespace);
+
+  // Install the license key into the namespace
+const licenseSecret = new k8s.yaml.ConfigGroup("license-secret",
+  { files: "/Users/debasishghosh/lightbend/akka-cloud-platform-gcp-deploy/license.yaml" },
```

Comment:
I think this needs to be part of configuration. But unlike other configuration parameters this cannot have any default value. Hence I am thinking of making this mandatory to be set from the cli before running pulumi. So, if the value is not set, pulumi will give the following error at startup ..

>pulumi:pulumi:Stack (akka-cloud-platform-gcp-deploy-dev):
    error: Missing required configuration variable 'akka-cloud-platform-gcp-deploy:license-file-path'
    	please set a value using the command `pulumi config set akka-cloud-platform-gcp-deploy:license-file-path <value>`

This looks to be a very user-friendly message with the command that needs to be invoked being part of the message.

Suggestions ? @RayRoestenburg @marcospereira @andreaTP

Reply:
Sounds good to me!

Reply:
implemented the plan outlined above